homeassistant/components/tplink/__init__.py

```diff
@@ -96,16 +99,23 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up TPLink from a config entry."""
     config_data = hass.data[DOMAIN].get(ATTR_CONFIG)
+    if config_data is None and entry.data is not None:
```

Comment:
When would `entry.data` be `None`?

Reply:
when the config entry is created first time by
https://github.com/home-assistant/core/blob/89f7cee1c53317365fb6ddbc993828c4ee0eae10/homeassistant/components/tplink/__init__.py#L90-L92

Reply:
The `ConfigEntry.data` is of type MappingProxyType:
https://github.com/home-assistant/core/blob/172769d19caf1c34480d1a31899f968ab0a5bfa0/homeassistant/config_entries.py#L194

```suggestion
    if config_data is None and entry.data:
```

Reply:
ahhh ... thx for that hint 🙂